cmd/vault-subpath-proxy/kv_update_transport.go

```diff
@@ -36,6 +42,16 @@ type kvUpdateTransport struct {
 	// If enabled, the roundtripper will wait for secret
 	// sync to complete. Should only be enabled in tests.
 	synchronousSecretSync bool
+
+	privilegedVaultClient                 *vaultclient.VaultClient
+	existingSecretKeysByNamespaceName     map[types.NamespacedName]sets.String
+	existingSecretKeysByNamespaceNameLock sync.Mutex
+	existingSecretKeysByVaultSecretName   map[string][]namespacedNameKey
```

Comment:
The duplicate data exists because:

* For validation of requests we need to index by Kubernetes secret
* But on update/delete we need to know all values by vault path - We could fetch it but that adds another api call and makes the code more complicated, because we have to handle the case that the user request fails and we have to revert our cache


Hypothetically we could store this in one data structure, but that would require to iterate everything for either validation or cache update.

Reply:
It's worth adding comments indicating where each are used and that `ByVaultSecretName` is used as an index into `ByNamespaceName`.

Reply:
Done